sentry-opentelemetry/sentry-opentelemetry-bootstrap/src/main/java/io/sentry/opentelemetry/SentryContextWrapper.java

```diff
@@ -20,9 +20,27 @@ private SentryContextWrapper(final @NotNull Context delegate) {
     this.delegate = delegate;
   }
 
+  @SuppressWarnings("unchecked")
   @Override
-  public <V> V get(final @NotNull ContextKey<V> contextKey) {
-    return delegate.get(contextKey);
+  public <V> @Nullable V get(final @NotNull ContextKey<V> contextKey) {
+    V result = delegate.get(contextKey);
+    if (Sentry.isGlobalHubMode()) {
+      if (result == null
+          || (result instanceof Span && !((Span) result).getSpanContext().isValid())) {
+        if (isOpentelemetrySpan(contextKey)) {
```

Comment:
Played around with this too much. It should check whether the key refers to a `Span` earlier.